conan/tools/apple/xcodedeps.py

```diff
@@ -105,6 +105,10 @@ class XcodeDeps(object):
         """)
 
     def __init__(self, conanfile):
+        if self.__class__.__name__ in conanfile.generators:
+            raise ConanException(f"{self.__class__.__name__} is declared in the generators "
```

Comment:
maybe it's time to start defining some more specific exception classes, instead of just using vague/generic `ConanException` for everything. IMO it should greatly improve debugging and troubleshooting.
maybe something like `GeneratorException` fits here.

Reply:
Only if at some place the new exception types are captured with a specific purpose and differently to other ``ConanException`` exceptions.

Reply:
yes, that's one of the reasons. right now, looking at PR itself, it's pretty hard to tell where exception is captured, if it's captured at all.

Reply:
The point of this exception is not to be captured other than to be displayed to the user as an error, right? So it would make little sense to customize

Reply:
Yes, no need to be captured, no special handling rather than showing an error message, this is why a ``ConanException`` is good enough.